scripts/docs/json_builder.py

```diff
@@ -0,0 +1,223 @@
+import json
+import os
+
+import pkgutil
+import importlib
+import sys
+from importlib.machinery import ModuleSpec
+from types import ModuleType, FunctionType
+import inspect
+from typing import Dict
+import re
+
+import docstring_parser
+from docstring_parser import parse
+from docstring_parser.common import DocstringStyle
+
+LOG = open("log.txt", "w+")
+
+
+def docstring_fixes(docstr: str) -> str:
+    docstr = re.sub(r"Args:[\n ]*None", "", docstr)  # Fix for Args: None in docstrings, which is not valid
+    return docstr
+
+
+def docstring_data(docstr: str):
+    if docstr is None:
+        return {}
+    docstr = docstring_fixes(docstr)
+    try:
+        parsed_docstr = parse(docstr, style=DocstringStyle.GOOGLE)
+    except docstring_parser.common.ParseError:
+        raise ValueError("Invalid docstring: " + docstr)
+    if ":param" in docstr:
+        raise ValueError("Invalid docstring (numpy): " + docstr)
```

Comment:
f-string

Reply:
The Google parser is "able" to parse numpy dosctrings. It parses the numpy docstr as plain text and ignores the directives like `:param`. A more robust solution of detecting the numpy format could be made, but I am not sure if it's worth the effort.  

I should definetely add a comment though